modules/core/src/main/java/org/apache/ignite/internal/pagemem/wal/record/delta/MetaPageUpdatePartitionDataRecordV2.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.internal.pagemem.wal.record.delta;
+
+import java.io.DataInput;
+import java.io.IOException;
+import java.nio.ByteBuffer;
+import org.apache.ignite.IgniteCheckedException;
+import org.apache.ignite.internal.pagemem.PageIdUtils;
+import org.apache.ignite.internal.pagemem.PageMemory;
+import org.apache.ignite.internal.processors.cache.persistence.tree.io.PagePartitionMetaIO;
+import org.apache.ignite.internal.processors.cache.persistence.tree.io.PagePartitionMetaIOV2;
+import org.apache.ignite.internal.util.typedef.internal.S;
+
+/**
+ *
+ */
+public class MetaPageUpdatePartitionDataRecordV2 extends MetaPageUpdatePartitionDataRecord {
+    /** */
+    private long link;
+
+    /**
+     * @param grpId Group id.
+     * @param pageId Page id.
+     * @param updateCntr Update counter.
+     * @param globalRmvId Global remove id.
+     * @param partSize Partition size.
+     * @param cntrsPageId Cntrs page id.
+     * @param state State.
+     * @param allocatedIdxCandidate Allocated index candidate.
+     * @param link Link.
+     */
+    public MetaPageUpdatePartitionDataRecordV2(
+        int grpId,
+        long pageId,
+        long updateCntr,
+        long globalRmvId,
+        int partSize,
+        long cntrsPageId,
+        byte state,
+        int allocatedIdxCandidate,
```

Comment:
I think it's not necessary - allocatedIdxCandidate was moved from meta page.

Reply:
This field is present in master. Better to remote it by separate ticket.